File: Data Structures & Algorithms/meeting-schedule/submission-1.py
"""
Definition of Interval:
class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end
"""

class Solution:
    def canAttendMeetings(self, intervals: List[Interval]) -> bool:
        if not intervals:
            return True
        intervals.sort(key = lambda i: i.start)
        ans = [intervals[0]]
        cur = 1
        while cur<len(intervals):
            if intervals[cur].start < ans[-1].end:
                return False
            else:
                ans.append(intervals[cur])
            cur += 1
        return True

File: Data Structures & Algorithms/boats-to-save-people/submission-1.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        int ans = 0, l = 0, r = people.size()-1;
        sort(people.begin(), people.end());
        while (l<=r){
            if (people[l]+people[r] <= limit){
                ++l;
            } 
            ++ans; 
            --r;
        }
        return ans;
    }
};